model/reliability/db.go

```diff
@@ -26,8 +26,72 @@ import (
 	"go.mongodb.org/mongo-driver/bson"
 )
 
+// DateBoundaries returns the date boundaries when splitting the period between 'start' and 'end' in groups of 'numDays' days.
+// The boundaries are the start dates of the periods of 'numDays' (or less for the last period), starting with 'start'.
+func (filter TaskReliabilityFilter) DateBoundaries() []time.Time {
```

Comment:
I'm a little suspicious of this not being a pointer receiver?

Reply:
It was originally a function rather than a method, but it does modify the struct state.

I had toyed with caching the result in a struct field but it only ended up being called once.

Reply:
if anything inside of DateBoundaries modifies filter then that won't be visable to callers (because unless the method is on a pointer, "filter" is a copy).